File: runme/mcgen/local/datan/exasrc/e1ds.c
#include <stdio.h>
#include "datsrc.h"

main(void)
{
    /* Local variables */
    static integer iexp, nfly, nexp, na, it, nt, nnfly;
    static double p, r[1000], ha;
    /* System generated locals */
    integer i__1, i__2;
/* simulates Example 5.1 */
/* identify program to user */
    printf("%s\n", "Program E1DS simulates empirical frequency");
    printf("%s\n\n", "and demonstrates statistical fluctuations.");
/* ask for parameters */
    printf("%s\n", "Enter number of flies NFLY (>0):");
    printf("%s", "> ");
    scanf("%li", &nfly);
    printf("%s\n", "Enter number of experiments NEXP (>0):");
    printf("%s", "> ");
    scanf("%li", &nexp);
    printf("%s\n", "Enter probability P(A), (0.<P(A)<1.):");
    printf("%s", "> ");
    scanf("%lf", &p);
/* loop over all experiments */
    i__1 = nexp;
    for (iexp = 1; iexp <= i__1; ++iexp) {
	nnfly = nfly;
	na = 0;
/* loop over all flies */
L10:
	nt = min(nnfly,1000);
	rnecuy_(r, &nt);
	i__2 = nt;
	for (it = 1; it <= i__2; ++it) {
	    if (r[it - 1] < p) {
		++na;
	    }
	}
	nnfly -= nt;
	if (nnfly > 0) {
	    goto L10;
	} else {
	    ha = (double) na / (double) nfly;
            printf("%s%3li%s%5li%s%10.8lf\n",
                   " IEXP = ", iexp, " ,  NA = ", na, " ,  HA = ", ha);
	}
    }
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e1ds.f
      PROGRAM E1DS
C simulates Example 5.1
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      DIMENSION R(1000)
C identify program to user
      WRITE(*,*)' Program E1DS simulates empirical frequency'
      WRITE(*,*)' and demonstrates statistical fluctuations'
      WRITE(*,*)' '
C ask for parameters
      WRITE(*,*)' Enter number of flies NFLY (>0)'
      WRITE(*,*)' >'
      READ(*,*) NFLY
      WRITE(*,*)' Enter number of experiments NEXP (>0)'
      WRITE(*,*)' >'
      READ(*,*) NEXP
      WRITE(*,*)' Enter probability P(A), (0.<P(A)<1.)'
      WRITE(*,*)' >'
      READ(*,*) P
C loop over all experiments
      DO 30 IEXP=1,NEXP
        NNFLY=NFLY
        NA=0
C loop over all flies
10      NT=MIN(NNFLY,1000)
        CALL RNECUY(R,NT)
        DO 20 IT=1,NT
          IF(R(IT).LT.P) NA=NA+1
20      CONTINUE
        NNFLY=NNFLY-NT
        IF(NNFLY.GT.0) THEN
          GO TO 10
        ELSE
          HA=DBLE(NA)/DBLE(NFLY)
          WRITE(*,1000)
     *    ' IEXP = ',IEXP,' ,  NA = ',NA,' ,  HA = ',HA
1000      FORMAT(1X,A,I3,A,I5,A,F10.8)
        END IF
30    CONTINUE
      END
